src/daft-core/src/array/ops/minhash.rs

```diff
@@ -0,0 +1,99 @@
+use std::{cmp, iter::repeat_with};
+
+use arrow2::array::{MutableArray, MutablePrimitiveArray, PrimitiveArray};
+use common_error::{DaftError, DaftResult};
+use mur3::murmurhash3_x86_32;
+
+use crate::{
+    array::FixedSizeListArray,
+    datatypes::{Field, Utf8Array},
+    DataType, Series,
+};
+
+use super::{as_arrow::AsArrow, DaftMinHash};
+
+const MERSENNE_PRIME: u64 = (1 << 61) - 1;
+const MAX_HASH: u32 = 0xffffffff;
+const DEFAULT_SEED: u32 = 1;
+
+impl DaftMinHash for Utf8Array {
+    type Output = DaftResult<FixedSizeListArray>;
+
+    fn minhash(&self, num_hashes: usize, ngram_size: usize, seed: Option<u32>) -> Self::Output {
+        if num_hashes == 0 {
+            return Err(DaftError::ValueError(
+                "Number of hashes must be nonzero".into(),
+            ));
+        }
+        if ngram_size == 0 {
+            return Err(DaftError::ValueError("Ngram size must be nonzero".into()));
+        }
+
+        // generate permutations
+        let seed = seed.unwrap_or(DEFAULT_SEED);
+        let mut rng = fastrand::Rng::with_seed(seed as u64);
+        let permutations: Vec<(u32, u32)> =
+            repeat_with(|| (rng.u32(1..=u32::MAX), rng.u32(1..=u32::MAX)))
```

Comment:
the max random number should probably be `i32::MAX` otherwise if you could have an overflow even with a u64.

`2**32 * 2**32 + 2**32` will overflow but
`2**32 * 2**31 + 2**32` will not.